### PR TITLE
Support PHP 8.2, drop support for PHP 8.0, upgrade to vimeo/psalm v5

### PR DESCRIPTION
**Pull request by internalsystemerror**
This PR:
- Adds support for PHP 8.2
- Adds PHP 8.2 to the matrix for testing
- Drops support for PHP 8.0
- Updates the locked php version to 8.1
- Updates `vimeo/psalm` to v5
- Updates `psalm/plugin-phpunit` to v0.18

Closes:
- https://github.com/Roave/psr-container-doctrine/pull/87
- https://github.com/Roave/psr-container-doctrine/pull/88

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ jobs:
         php-version:
           - '8.0'
```
Comment (Ocramius) [inline .github/workflows/test.yml]: Should be dropped, no?

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by internalsystemerror** [comment]
These psalm errors don't show up locally, and I can fix all but 1, which  gives an error locally OR on CI, but I can't solve for both:

https://github.com/Roave/psr-container-doctrine/actions/runs/3664302981/jobs/6194637843#step:6:28

When fixed for CI, locally I get a warning about an invalid suppresion.

**Comment by Ocramius** [comment]
Maybe that should just be a `Memcached::class` reference?

**Comment by internalsystemerror** [comment]
Done. I was trying to avoid importing `memcached` as that's why the suppression was originally there. But it only needs importing into the test it's used.

**Comment by internalsystemerror** [comment]
Oh, and memcached isn't actually installed in CI, so the builds will fail :/

**Comment by Ocramius** [comment]
Wait... PHP 8.1 now fails installation? D:

**Comment by internalsystemerror** [comment]
Yeah, I've noticed now that memcached was disabled in `.github/workflows/test.yml` waiting on a fix for https://github.com/php-memcached-dev/php-memcached/issues/477 ... Let me revert some stuff here and put back the original suppression ... It fails locally, but that might be my setup.

**Comment by internalsystemerror** [comment]
OK ... Try that, should be all green now.

**Review by Ocramius — APPROVED** [review]
Awesome, thanks!

**Pull request merged**